File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String input = br.readLine().strip(); 
        String target = "UCPC";
        int idx = 0;

        for (int i = 0; i < input.length(); i++) {
            if (input.charAt(i) == target.charAt(idx)) {
                idx++;
                if (idx == target.length()) { 
                    System.out.println("I love UCPC");
                    return;
                }
            }
        }
        System.out.println("I hate UCPC");
    }
}

File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.py
import sys
input = sys.stdin.readline

target ='UCPC'
world = input().strip()
idx = 0

for char in world:
    if char==target[idx]:
        idx+=1
        if idx == len(target):
            print("I love UCPC")
            break
else:
    print("I hate UCPC")